src/renderer/components/server-modal-form.jsx

```diff
@@ -130,6 +131,15 @@ export default class ServerModalForm extends Component {
     return hasError ? 'error' : '';
   }
 
+  handleOnClientChange(client) {
+    this.setState({ client });
+
+    const clientConfig = CLIENTS.find(entry => entry.value === client);
+    if (clientConfig && clientConfig.defaultPort) {
```

Comment:
i thought about that... i have even added that... but imagine that you are editing one connection you have added mysql but actually it was pgsql... should we disallow default port?

for me it makes sense too to not change when have state but while i was testing i found it a bit weird...